Add deprecation warning for lora_extra_vocab_size

ahengljh opened the pull request:
<!-- markdownlint-disable -->
## Purpose

Add deprecation warning for `lora_extra_vocab_size` parameter as suggested in PR #23540. This is a gentler approach than immediate removal, giving users time to adapt their code.

Related to: #23474

## Test Plan

Test that the deprecation warning appears when using LoRA with non-zero `lora_extra_vocab_size`:

```python
from vllm.config import LoRAConfig
config = LoRAConfig()
config = LoRAConfig(lora_extra_vocab_size=256)
```

## Test Result

```
>>> from vllm.config import LoRAConfig
INFO 08-26 16:37:46 [__init__.py:241] Automatically detected platform cuda.
>>> config = LoRAConfig()
WARNING 08-26 16:38:01 [__init__.py:2502] `lora_extra_vocab_size` is deprecated and will be removed in a future release. Additional vocabulary support for LoRA adapters is being phased out.
```

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

ahengljh committed:
Add deprecation warning for lora_extra_vocab_size

Signed-off-by: Jinheng Li <ahengljh@gmail.com>

jeejeelee commented:
Please fix the pre-cmmit failure

ahengljh committed:
Specify v0.12.0 removal for lora_extra_vocab_size deprecation

Signed-off-by: Jinheng Li <ahengljh@gmail.com>

ahengljh commented:
> Please fix the pre-cmmit failure

Done. It got passed.

hmellor commented:
fastcheck is not necessary to merge a PR, I'll enable the main tests to see if this broke anything

hmellor reviewed (changes requested):
Looks like a bunch of LoRA tests are now failing. Could you fix the tests to work with the new default?

ahengljh committed:
Fix test failure by keeping lora_extra_vocab_size default at 256

The previous commit changed the default value from 256 to 0, which
caused test_v1_llm_by_default to fail. This commit keeps the original
default value of 256 while still adding the deprecation warning to
notify users that this feature will be removed in v0.12.0.

The deprecation warning will always be shown when LoRAConfig is created,
alerting users to the upcoming removal without breaking existing code.

Signed-off-by: Jinheng Li <ahengljh@gmail.com>
